lib/Sema/TypeCheckConstraints.cpp

```diff
@@ -451,8 +451,20 @@ resolveDeclRefExpr(UnresolvedDeclRefExpr *UDRE, DeclContext *DC) {
 
     performTypoCorrection(DC, UDRE->getRefKind(), Type(), Name, Loc,
                           lookupOptions, Lookup);
+    
+    // Check if any lookups share a basename. If that's the case, we need
+    // to disambiguate the diagnostic by printing the UDRE's full name.
```

Comment:
DeclNames in diagnostics currently print with `skipEmptyArgumentNames=true`, which will print `foo(_:)` as `foo` because the only argument name is technically empty. I don't think this is the "right" behavior...